grand/simulation/antenna/tabulated.py

```diff
@@ -120,17 +132,67 @@ def _load_from_numpy(cls, path: Union[Path, str]) -> TabulatedAntennaModel:
                       leff_phi = leffp, phase_phi = phasep)
         return cls(table=t)
 
+    @classmethod
+    def _load_from_hdf5(cls, path: Union[Path, str], arm: str = "EW") -> TabulatedAntennaModel:
+        print("tuutu")
+        # Open the HDF5 file with antenna info
+        ant_file = h5py.File(path, 'r')
+        # Select the requested arm from the file
+        ant_arm = ant_file[arm]
+        # Load the antenna parameters (to numpy arrays, thus [:])
+        f = ant_arm["frequency"][:]
+        R = ant_arm["resistance"][:]
+        X = ant_arm["reactance"][:]
+        theta = ant_arm["theta"][:]
+        phi = ant_arm["phi"][:]
+        lefft = ant_arm["leff_theta"][:]
+        leffp = ant_arm["leff_phi"][:]
+        phaset = ant_arm["phase_theta"][:]
+        phasep = ant_arm["phase_phi"][:]
+        print("all shape", f.shape, R.shape, X.shape, theta.shape, phi.shape, lefft.shape, leffp.shape, phaset.shape, phasep.shape)
+        
+        n_f = f.shape[0]
+        n_theta = len(numpy.unique(theta[0,:]))
+        n_phi = int(R.shape[1] / n_theta)
+        shape = (n_f, n_phi, n_theta)
+
+        dtype = 'f4'
+        f = f[:,0].astype(dtype) * u.MHz
+        theta = theta[0, :n_theta].astype(dtype) * u.deg
+        phi = phi[0, ::n_theta].astype(dtype) * u.deg
+        R = R.reshape(shape).astype(dtype) * u.Ohm
+        X = X.reshape(shape).astype(dtype) * u.Ohm
+        lefft = lefft.reshape(shape).astype(dtype) * u.m
+        leffp = leffp.reshape(shape).astype(dtype) * u.m
+        phaset = numpy.deg2rad(phaset).reshape(shape).astype(dtype) * u.rad
+        phasep = numpy.deg2rad(phasep).reshape(shape).astype(dtype) * u.rad
+
+        t = DataTable(frequency = f, theta = theta, phi = phi, resistance = R,
+                      reactance = X, leff_theta = lefft, phase_theta = phaset,
+                      leff_phi = leffp, phase_phi = phasep)
+        return cls(table=t)
+
+
     def effective_length(self, direction: BaseRepresentation,
         frequency: u.Quantity) -> CartesianRepresentation:
-
-        direction = direction.represent_as(PhysicsSphericalRepresentation)
-        theta, phi = direction.theta, direction.phi
+        
+        # LWP: Replace with manual conversion to spherical, assuming direction is a numpy array
+        if grand_astropy:
+            direction = direction.represent_as(PhysicsSphericalRepresentation)
+            theta, phi = direction.theta, direction.phi
+        else:
+            theta = numpy.arctan2(numpy.sqrt(direction[0]**2+direction[1]**2),direction[2])
+            phi = numpy.arctan2(direction[1],direction[0])
```

Comment:
Indee, this would be enough for now. With the proposed coordinates type extending numpy.ndarray it could be something like:
```python
spherical = direction.parametrize_as(SphericalVector)
theta, phi = spherical.theta, spherical.phi
```